### PR TITLE
RVIC 1.1 release merge from develop

### PR DESCRIPTION
**Pull request by jhamman**
RVIC 1.1 updates, still waiting on:
- [x] handling of missing values in forcing fields #53.
- [x] fix multiprocessing #50
- [x] test and/or fixes for `remap==False` #36
- [x] fix monthly means bug
- [x] add ~~Stehekin~~ RASM and PNW sample dataset to repository

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -103,12 +103,12 @@ def write_version_py(filename=None):
       author='Joe Hamman',
       author_email='jhamman@hydro.washington.edu',
```
Comment (yixinmao) [inline setup.py]: I guess your email now is uw.edu one?

File: rvic/parameters.py [hunk 2]
```diff
@@ -319,14 +335,13 @@ def gen_uh_run(uh_box, fdr_data, fdr_vatts, dom_data, outlet, config_dict,
     """
```
Comment (yixinmao) [inline rvic/parameters.py]: This is just a general comment - maybe you can add more detailed descriptions for each function about what it is doing and what its input/output variables and their type/structure are? For each piece of code - maybe add also add more comments about their purpose? For major variables - maybe add their data type/dimension/etc.?

File: rvic/core/history.py [hunk 3]
```diff
@@ -646,27 +671,27 @@ def __write_array(self):
         outlet_decomp_ind[:] = self._outlet_decomp_ind
         onm[:, :] = char_names
 
-        for key, val in share.outlet_lon.__dict__.iteritems():
+        for key, val in iteritems(share.outlet_lon.__dict__):
```
Comment (jhamman) [inline rvic/core/history.py]: I don't think we need the `__dict__` portion here. Same for all instances below.

File: rvic/core/aggregate.py [hunk 4]
```diff
@@ -1,15 +1,15 @@
-#!/usr/local/bin/python
```
Comment (orianac) [inline rvic/core/aggregate.py]: You don't need this line but you need the utf line below?

File: rvic/core/config.py [hunk 5]
```diff
@@ -84,7 +84,7 @@ def config_type(value):
 def isfloat(x):
     """Test of value is a float"""
```
Comment (orianac) [inline rvic/core/config.py]: "if" not "of"

File: rvic/convert.py [hunk 6]
```diff
@@ -70,17 +72,16 @@ def uhs2param_init(config_file):
                       options['VERBOSE'])
 
     for direc in directories:
-        log.info('%s directory is %s' % (direc, directories[direc]))
+        log.info('%s directory is %s', direc, directories[direc])
```
Comment (yixinmao) [inline rvic/convert.py]: Better to use .format()?

File: rvic/convert.py [hunk 7]
```diff
@@ -70,17 +72,16 @@ def uhs2param_init(config_file):
...
     # ---------------------------------------------------------------- #
 
     # ---------------------------------------------------------------- #
     # Read domain file (if applicable)
-    dom_data, DomVats, DomGats = read_domain(config_dict['DOMAIN'])
-    log.info('Opened Domain File: %s' % config_dict['DOMAIN']['FILE_NAME'])
+    dom_data = read_domain(config_dict['DOMAIN'])[0]
+    log.info('Opened Domain File: %s', config_dict['DOMAIN']['FILE_NAME'])
```
Comment (yixinmao) [inline rvic/convert.py]: Better to use .format()?

File: rvic/convert.py [hunk 8]
```diff
@@ -70,17 +72,16 @@ def uhs2param_init(config_file):
...
 
     if 'NEW_DOMAIN' in config_dict:
-        new_dom_data, new_DomVats, \
-            new_DomGats = read_domain(config_dict['NEW_DOMAIN'])
+        new_dom_data = read_domain(config_dict['NEW_DOMAIN'])[0]
         log.info('Opened New Domain File: %s',
                  config_dict['NEW_DOMAIN']['FILE_NAME'])
```
Comment (yixinmao) [inline rvic/convert.py]: Better to use .format()?

File: rvic/convolution.py [hunk 9]
```diff
@@ -83,7 +84,7 @@ def convolution_init(config_file):
 
     # ---------------------------------------------------------------- #
     # Copy Inputs to $case_dir/inputs and update configuration
-    #config_dict = copy_inputs(config_file, directories['inputs'])
+    # config_dict = copy_inputs(config_file, directories['inputs'])
```
Comment (yixinmao) [inline rvic/convolution.py]: This commented line is not necessary?

File: rvic/core/read_forcing.py [hunk 10]
```diff
@@ -1,3 +1,4 @@
+# -*- coding: utf-8 -*-
 """
 read_forcings.py
```
Comment (orianac) [inline rvic/core/read_forcing.py]: Maybe a more descriptive piece of text here?

File: rvic/core/share.py [hunk 11]
```diff
@@ -91,68 +92,72 @@ def __init__(self,
 
         self.atts = OrderedDict()
 
-        if title:
+        if title is not None:
```
Comment (orianac) [inline rvic/core/share.py]: Why make this change?

File: tests/run_tests.py [hunk 12]
```diff
@@ -81,30 +84,52 @@ def run_examples(config_file):
...
-                                  numofproc=numofproc)
+            try:
+                parameters.parameters(test_dict['config_file'],
+                                      numofproc=numofproc)
+                test_outcomes[test] = 'Passed'
+            except Exception as e:
+                print('Error in parameters example: {0}'.format(test))
+                test_outcomes[test] = 'Failed: {0}'.format(e)
         else:
             raise ValueError('Unknow function variable: '
```
Comment (orianac) [inline tests/run_tests.py]: "Unknown"

File: tests/unit/test_share.py [hunk 13]
```diff
@@ -0,0 +1,11 @@
+from rvic.core.share import NcGlobals, NcVar
+
+
+def test_nc_globals():
+    ncg = NcGlobals()
+    ncg.update()
+
+
+def test_nc_var():
+    ncv = NcVar(test='test')
```
Comment (orianac) [inline tests/unit/test_share.py]: add spaces? Or maybe not... hmmmmmmm

File: rvic/core/aggregate.py (hunk 4 above)
Comment (jhamman) [inline rvic/core/aggregate.py]: correct.

File: rvic/core/config.py (hunk 5 above)
Comment (jhamman) [inline rvic/core/config.py]: if - yes. I don't think it is worth adding a dependency for a simple function like this.

File: rvic/convert.py (hunk 6 above)
Comment (jhamman) [inline rvic/convert.py]: no.  the `logging` module and `log` objects will expand the strings automatically.

File: docs/user-guide/install.md [hunk 14]
```diff
@@ -1,10 +1,11 @@
 # Installing RVIC
 
 ## Dependencies
-- [Python 2.7](http://www.python.org/)
+- [Python 2.7 or later including Python3](http://www.python.org/)
 - [Numpy](http://www.numpy.org)
 - [Scipy](http://www.scipy.org/)
 - [netcdf4-python](https://code.google.com/p/netcdf4-python/)
+- [Pandas](http://pandas.pydata.org/)
```
Comment (bartnijssen) [inline docs/user-guide/install.md]: This and previous lines - if you want to be consistent to with how the groups capitalize their packages, then I think it should be:
 * NumPy
 * SciPy
 * pandas

Also, I think you can remove the trailing `/` from all the links

File: tests/unit/test_share.py (hunk 13 above)
Comment (jhamman) [inline tests/unit/test_share.py]: not for keyword args...:https://www.python.org/dev/peps/pep-0008/#id21

File: rvic/convert.py [hunk 15]
```diff
@@ -1,19 +1,20 @@
...
-from core.utilities import tar_inputs
-from core.convert import read_station_file, read_uhs_files, move_domain
-from core.param_file import finish_params
-from core.config import read_config
+from .core.log import init_logger, close_logger, LOG_NAME
+from .core.utilities import make_directories, copy_inputs, read_domain
+from .core.utilities import tar_inputs
+from .core.convert import read_station_file, read_uhs_files, move_domain
+from .core.param_file import finish_params
+from .core.config import read_config
```
Comment (bartnijssen) [inline rvic/convert.py]: why `.core` rather than core?

File: rvic/convert.py (hunk 6 above)
Comment (bartnijssen) [inline rvic/convert.py]: If you are changing these anyway, you may want to consider python type string formatting, i.e.:

```python
log.info('{} directory is {}'.format(direc, directories[direc]))
```

File: rvic/core/share.py (hunk 11 above)
Comment (jhamman) [inline rvic/core/share.py]: `is not None` is more specific. This allows title to be `0`, or `False, or `''` while continuing on...

**Comment by bartnijssen** [comment]
Since all these have been reviewed, go ahead and merge into release yourself.

**Pull request merged**